data/sea/54-zebra.h

```diff
@@ -154,68 +190,125 @@ static ierror_msg_t zebra_translate
     return 0;
 }
 
-static int64_t zebra_translate_table
+/* Translate Zebra enum tags to Icicle bool flags, e.g. is_some, is_right, etc.
+   We only allow at most two variants since Icicle's runtime representation
+   of enums is unsuitable for more.
+ */
+static int64_t zebra_translate_column_tags
+    ( const int64_t variant_count
+    , void **dst
+    , const int64_t *src )
+{
+    if (variant_count > 2) {
+        fprintf (stderr, "Fatal error: found an enum with %lld variants. Icicle does not support Zebra enums with more than two variants.\n", variant_count);
+        exit(1);
+    }
+
+    *dst = src;
+    return 1;
+}
+
+static int64_t zebra_translate_column_nested
     ( anemone_mempool_t *mempool
-    , iint_t elem_start
-    , iint_t count
+    , int64_t elem_start
+    , int64_t elem_count
     , void **dst
-    , const zebra_table_t *table )
+    , const int64_t *indices
+    , const zebra_table_t *src )
 {
-    int64_t         offset = 0;
-    int64_t         cols_count  = table->column_count;
-    zebra_column_t *cols = table->columns;
-
-    //printf ("zebra_translate_table: elem_start = %d, count = %d\n", elem_start, count);
-    for (int64_t i = 0; i != cols_count; ++i) {
-        //printf ("zebra_translate_table: offset = %d\n", offset);
-        offset += zebra_translate_column (mempool, elem_start, count, dst + offset, cols + i);
+    /* Only read nested tables of type binary now.
+       FIXME: slice out nested tables, when we need to support them. */
+    if (src->tag != ZEBRA_TABLE_BINARY) {
+        fprintf (stderr, "Fatal error: found a nested table with tag %lld. Icicle only supports nested binary table for now.\n", src->tag);
+        exit(1);
     }
 
+    int64_t offset = 1;
+
+    for (int i = 0; i != elem_count; ++i) {
+        const int64_t elem_index = elem_start + i;
+        const int64_t table_start = indices[elem_index] - indices[0];
+        const int64_t table_row_count = indices[elem_index + 1] - indices[elem_index];
+
+        /* read the inner tables */
+        void **target = anemone_mempool_alloc (mempool, elem_count * 8);
```

Comment:
yeah it should... this is just allocating the pointers for the nested table, right now we only have the binary case anyway